### ISSUE TITLE
oas3-valid-schema-example Rule passes with empty examples array

### ISSUE THREAD

**Issue by yashwason** [daveshanley/vacuum#345]
I have the `oas3-valid-schema-example` enabled on my Vacuum configuration. It rightly warns about missing examples on fields. But I noticed the warning goes away if I provide an empty examples array too.

```
                status:
                    type: integer
                    description: The HTTP status code for the error.
                    examples:
                        - 400
                        - 413
                detail:
                    type: string
                    description: A human-readable explanation of the error.
                instance:
                    type: string
                    description: A unique identifier for this particular occurrence of the error.
                    examples:
                      - 
```

When linting the above excerpt, `status` rightfully passes, `detail` rightfully fails; but `instance` passes even though it should not.

**Comment by daveshanley** [comment]
interesting edge case. vacuum sees the examples with an array (even though its empty) and is skipping over doing anymore work here because there is nothing to check, but it's still marking it as good. 

Looks like a bug for sure, will pick up on my next bug sweep.

**Comment by daveshanley** [comment]
This has been resolved in `v0.7.0`, all the the example logic was re-written and this case was covered as a part of that rewrite.
